tmt/steps/execute/internal.py

```diff
@@ -167,13 +167,15 @@ def log(key, value=None, color=None, shift=1, level=1):
     def check(self, test):
         """ Check the test result """
         self.debug(f"Check result of '{test.name}'.")
+        if test.result == 'custom':
+            return self.check_custom_results(test)
         if test.framework == 'beakerlib':
-            return self.check_beakerlib(test)
+            return [self.check_beakerlib(test)]
         else:
             try:
-                return self.check_result_file(test)
+                return [self.check_result_file(test)]
             except tmt.utils.FileError:
-                return self.check_shell(test)
+                return [self.check_shell(test)]
```

Comment:
I'd suggest changing other methods to return a list as well, even if it'd contain a single item. This way, all methods `check_{beakerlib,shell,result_file,custom}` would present consistent return value type.

Reply:
Done